sentry/src/main/java/io/sentry/internal/SpotlightIntegration.java

```diff
@@ -0,0 +1,119 @@
+package io.sentry.internal;
+
+import static io.sentry.SentryLevel.DEBUG;
+import static io.sentry.SentryLevel.ERROR;
+
+import io.sentry.Hint;
+import io.sentry.IHub;
+import io.sentry.Integration;
+import io.sentry.SentryEnvelope;
+import io.sentry.SentryOptions;
+import io.sentry.util.Objects;
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.zip.GZIPOutputStream;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+@ApiStatus.Internal
+public final class SpotlightIntegration
+    implements Integration, SentryOptions.BeforeEnvelopeCallback, Closeable {
+
+  private @NotNull SentryOptions options = SentryOptions.empty();
+
+  private final @NotNull ExecutorService executorService = Executors.newSingleThreadExecutor();
```

Comment:
`h` can we use executor service from options? This one doesn't set `daemon` and thus will likely block shutdown for backend (https://github.com/getsentry/sentry-java/pull/2747)

Reply:
I've switched to using our `SentryExecutorService`, which is setup in daemon mode. I still wouldn't use the executor service from our options, as it could slow down any disk/cpu bound tasks when many envelopes are queuing up.

Reply:
@adinauer would the mentioned `SentryExecutorService` work fine for java backend use cases?

Reply:
Do you mean the one from options or a new one? Either should work. Is there a specific reason you're asking?
Can't we increase the number of threads `SentryExecutorService ` is allowed to use in case we run into problems?

There's `Executors.newScheduledThreadPool` which we could pass a pool size to.

Reply:
Afaik we can't increase the pool size for `options.executorService`. At least not on Android, right @romtsn? IIRC on SDK init we rely on some background tasks to be executed in a certain order.